src/Npgsql/SqlQueryParser.cs

```diff
@@ -61,9 +55,9 @@ class SqlQueryParser
             if (currCharOfs >= end)
                 goto Finish;
             var lastChar = ch;
-            ch = sql[currCharOfs++];
+            ch = span[currCharOfs++];
         NoneContinue:
-            for (; ; lastChar = ch, ch = sql[currCharOfs++])
+            for (; ; lastChar = ch, ch = span[currCharOfs++])
```

Comment:
I added `when` suggestions below but I realize many of the if x then goto else break cases could use it.

Reply:
True